pkg/machineoutput/types.go

```diff
@@ -30,6 +31,26 @@ type GenericSuccess struct {
 	Message           string `json:"message"`
 }
 
+// unindentedMutex prevents multiple JSON objects from being outputted simultaneously on the same line. This is only
+// required for OutputSuccessUnindented's 'unindented' JSON objects, since objects printed by other methods are not written from
+// multiple threads.
+var unindentedMutex = &sync.Mutex{}
+
+// OutputSuccessUnindented outputs a "successful" machine-readable output format in unindented json
+func OutputSuccessUnindented(machineOutput interface{}) {
+	printableOutput, err := json.Marshal(machineOutput)
+
+	unindentedMutex.Lock()
+	defer unindentedMutex.Unlock()
+
+	// If we error out... there's no way to output it (since we disable logging when using -o json)
+	if err != nil {
+		fmt.Fprintf(log.GetStderr(), "Unable to unmarshal JSON: %s\n", err.Error())
```

Comment:
Is the `Unable to unmarshal JSON` error message necessary? Wouldn't that be contained in the error returned from `json.Marshal`?

Reply:
Also, could we use `log.Errorf`?

Reply:
These lines are copied directly from the OutputSuccess/OutputError functions that were already in that file, but re: `log.Errorf`, can't use it since output from all log.* functions is disabled (throughout the code) when `-o JSON` is specified. This is part of the original machine output design, where such an output would break the pretty-printed JSON output used by consuming tools.